### ISSUE TITLE
🐛 a11y warning on "link skills to experience" dialog

### ISSUE THREAD

**Issue by petertgiles** [GCTC-NTGC/gc-digital-talent#10864]
## 🐛 Bug

I noticed there's a warning about a missing description on the "link skills to experience dialog:
> Warning: Missing `Description` or `aria-describedby={undefined}` for {DialogContent}.

## 📋 Steps to Reproduce

1. log in
2. start an application
3. advance to the "skills requiremetns" step
4. link an experience to a skill using the dialog

## 📸 Screenshot

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/e2e2d269-50a9-430f-835d-c2cf1d4a4875)

## ✅ Acceptance Criteria

- [ ] no a11y warning when opening this dialog

**Comment by petertgiles** [comment]
Also noticed on the candidate assessment dialog.  Might affect many dialogs?
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/1a588eab-f08e-42c5-be61-6f12318cec31)

**Comment by vd1992** [comment]
Saw it in multiple places too in pool workflows (publish, changed closing, duplicate and so on...)

**Comment by esizer** [comment]
Looks like we just need to [add `aria-decsbribedby={undefined}` ](https://www.radix-ui.com/primitives/docs/components/dialog#description:~:text=If%20you%20want%20to%20hide%20the%20description%2C%20wrap%20it%20inside%20our%20Visually%20Hidden%20utility%20like%20this%20%3CVisuallyHidden%20asChild%3E.%20If%20you%20want%20to%20remove%20the%20description%20entirely%2C%20remove%20this%20part%20and%20pass%20aria%2Ddescribedby%3D%7Bundefined%7D%20to%20Dialog.Content.)to any dialogs that do not contain descriptions. 

Error message is worded in an odd way but their docs explain it a bit better :sweat_smile:

**Comment by brindasasi** [comment]
This is base component fix and should fix all of them.
